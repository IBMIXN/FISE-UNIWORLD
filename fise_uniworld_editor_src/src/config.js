import axios from "axios";

const axiosConfig = axios.create({
  baseURL: "https://uniworld.azurewebsites.net/api",
  headers: {
    "Content-type": "application/json",
  },
});

const platformURL = "https://brianmin.tech/uniworld";

export { axiosConfig, platformURL };
