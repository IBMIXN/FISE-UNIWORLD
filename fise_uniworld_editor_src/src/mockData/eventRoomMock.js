const mockEventRoom = {
  title: "mockTitle",
  eventDate: "2021-03-17T00:00:00.000Z",
  meetingTables: [],
  background: "mockBackground",
  scene: "mockScene",
};

const mockEventRoomUpdate = {
  title: "newMockTitle",
  eventDate: "2021-02-27T12:00:00.000Z",
  meetingTables: [],
  background: "newMockBackground",
  scene: "newMockScene",
};

const mockMeetingTable = {
  title: "mockTitle",
  type: "RoundMeetingTable",
  logoUrl: "https://via.placeholder.com/150",
  zoomUrl: "https://zoom.us/",
  posX: 0,
  posY: 0,
};

export { mockEventRoom, mockEventRoomUpdate, mockMeetingTable };
