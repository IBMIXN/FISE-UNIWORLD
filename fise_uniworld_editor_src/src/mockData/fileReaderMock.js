export default class FileReaderMock {
  onerror = jest.fn();
  onload = jest.fn();
  addEventListener = (_, fn) => fn({ target: { result: {} } });
  readAsDataURL = jest.fn();
}
