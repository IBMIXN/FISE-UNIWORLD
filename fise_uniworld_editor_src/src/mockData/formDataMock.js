export default class FormDataMock {
  append = jest.fn();
}
