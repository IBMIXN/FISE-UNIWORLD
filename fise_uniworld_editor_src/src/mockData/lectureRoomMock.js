const mockLectureRoom = {
  title: "mockTitle",
  module: "mockModule",
  lecturer: "mockLecturer",
  startTime: "2021-03-17T09:00:00.000Z",
  endTime: "2021-03-17T10:00:00.000Z",
  firstSlideUrl: "mockUrl",
  numSlides: 1,
};

const mockLectureRoomUpdate = {
  title: "newMockTitle",
  module: "newMockModule",
  lecturer: "newMockLecturer",
  startTime: "2021-02-27T14:00:00.000Z",
  endTime: "2021-02-27T15:00:00.000Z",
  firstSlideUrl: "newMockUrl",
  numSlides: 3,
};

export { mockLectureRoom, mockLectureRoomUpdate };
