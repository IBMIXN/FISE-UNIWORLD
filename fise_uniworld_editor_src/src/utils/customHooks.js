import React from "react";

const useReducerState = (defaultState) => {
  const reducer = (state, newState) => ({ ...state, ...newState });
  return React.useReducer(reducer, defaultState);
};

export { useReducerState };
