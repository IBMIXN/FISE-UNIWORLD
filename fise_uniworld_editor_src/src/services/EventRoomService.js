import { axiosConfig } from "../config";

class EventRoomService {
  url = "/eventrooms";

  async getAll() {
    return (await axiosConfig.get(`${this.url}?showDetails=true`)).data;
  }

  async get(id) {
    return (await axiosConfig.get(`${this.url}/${id}`)).data;
  }

  async create(data) {
    return (await axiosConfig.post(this.url, data)).data;
  }

  async update(id, data) {
    return (await axiosConfig.put(`${this.url}/${id}`, data)).data;
  }

  async delete(id) {
    return await axiosConfig.delete(`${this.url}/${id}`);
  }
}

export default new EventRoomService();
