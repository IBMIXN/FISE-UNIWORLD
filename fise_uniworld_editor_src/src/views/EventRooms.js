import React from "react";
import moment from "moment";
import { makeStyles } from "@material-ui/core/styles";
import {
  Grid,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  Container,
  Typography,
  Card,
  CardActions,
  CardContent,
  Button,
  CircularProgress,
  Snackbar,
} from "@material-ui/core";
import MuiAlert from "@material-ui/lab/Alert";
import DesktopWindowsIcon from "@material-ui/icons/DesktopWindows";
import AddIcon from "@material-ui/icons/Add";
import TableGrid from "../components/TableGrid";
import EventRoomService from "../services/EventRoomService";
import { Link as RouterLink } from "react-router-dom";
import { platformURL } from "../config";
import { useReducerState } from "../utils/customHooks";
import { findBackground, findScene, isBackgroundPreset } from "../utils/presets";
import { getFileNameFromBlobUrl } from "../utils/blobs";

const useStyles = makeStyles((theme) => ({
  mainSection: {
    paddingTop: theme.spacing(5),
  },
  eventInfoPanel: {
    marginTop: theme.spacing(1),
    padding: theme.spacing(2),
  },
  divider: {
    padding: theme.spacing(1),
  },
  centerTableGrid: {
    display: "flex",
    justifyContent: "center",
  },
  progressLoader: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    height: "50vh",
  },
  listItemCreate: {
    "&:hover": {
      backgroundColor: theme.palette.success.dark,
    },
    backgroundColor: theme.palette.success.main,
    color: "#fff",
  },
  mt: {
    marginTop: theme.spacing(1),
  },
}));

const EventRooms = (props) => {
  const classes = useStyles();
  const [eventRooms, setEventRooms] = React.useState([]);
  const [currentEventRoom, setCurrentEventRoom] = React.useState(0);
  const [isLoading, setLoading] = React.useState(false);
  const [alert, setAlert] = useReducerState({
    open: false,
    message: "",
    type: "success",
  });

  const showAlert = (msg, type) => {
    setAlert({ open: true, message: msg, type: type });
  };

  const hideAlert = () => {
    setAlert({ open: false });
  };

  React.useEffect(() => {
    if (props.location.state && props.location.state.created) {
      showAlert("Successfully created event room.", "success");
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [props.location]);

  React.useEffect(() => {
    async function fetchEventRooms() {
      try {
        setLoading(true);
        const data = await EventRoomService.getAll();
        if (props.match.params.id) {
          for (let i = 0; i < data.length; i++) {
            if (data[i]._id === props.match.params.id) {
              setCurrentEventRoom(i);
              break;
            }
          }
        }
        setEventRooms(data);
      } catch (err) {
        console.log(err);
      }
      setLoading(false);
    }
    fetchEventRooms();
  }, [props.match.params.id]);

  const handleClickEventRoom = (index) => {
    setCurrentEventRoom(index);
  };

  const handleClickRemove = async (id) => {
    if (window.confirm("Are you sure?")) {
      try {
        await EventRoomService.delete(id);
        setCurrentEventRoom(0);
        setEventRooms(eventRooms.filter((eventRoom) => eventRoom._id !== id));
      } catch (err) {
        console.log(err);
      }
    }
  };

  const handleClickEdit = (index) => {
    console.log(eventRooms[index]);
    props.history.push({
      pathname: "/edit",
      state: { eventRoom: eventRooms[index] },
    });
  };

  return (
    <Container className={classes.mainSection}>
      <Snackbar
        open={alert.open}
        autoHideDuration={3000}
        onClose={hideAlert}
        anchorOrigin={{ vertical: "top", horizontal: "center" }}
      >
        <MuiAlert elevation={6} variant="filled" onClose={hideAlert} severity={alert.type}>
          {alert.message}
        </MuiAlert>
      </Snackbar>
      <Typography variant="h4">Event Rooms</Typography>
      {isLoading ? (
        <div className={classes.progressLoader}>
          <CircularProgress />
        </div>
      ) : (
        <Grid container spacing={2}>
          <Grid item md={3} xs={12}>
            <List component="nav" aria-label="event rooms">
              <ListItem
                button
                component={RouterLink}
                to="/create"
                className={classes.listItemCreate}
              >
                <ListItemIcon>
                  <AddIcon style={{ color: "#fff" }} />
                </ListItemIcon>
                <ListItemText
                  primary="Create event room"
                  primaryTypographyProps={{ noWrap: true }}
                />
              </ListItem>
              {eventRooms.length > 0 &&
                eventRooms.map((eventRoom, index) => (
                  <ListItem
                    id={`listItem${index}`}
                    key={eventRoom._id}
                    button
                    selected={currentEventRoom === index}
                    onClick={() => handleClickEventRoom(index)}
                  >
                    <ListItemIcon>
                      <DesktopWindowsIcon />
                    </ListItemIcon>
                    <ListItemText
                      primary={eventRoom.title}
                      primaryTypographyProps={{ noWrap: true }}
                    />
                  </ListItem>
                ))}
            </List>
          </Grid>
          <Grid item md={9} xs={12}>
            <Card className={classes.eventInfoPanel}>
              {eventRooms.length > 0 && !!eventRooms[currentEventRoom] ? (
                <React.Fragment>
                  <CardContent>
                    <Typography color="textSecondary" gutterBottom>
                      {moment(eventRooms[currentEventRoom].eventDate).format(
                        "MMMM DD YYYY, h:mm a"
                      )}
                    </Typography>
                    <Typography id="currentEventRoomTitle" variant="h5" component="h2">
                      {eventRooms[currentEventRoom].title}
                    </Typography>
                    <div className={classes.divider} />
                    <Grid container>
                      <Grid item md={9} xs={12} className={classes.centerTableGrid}>
                        <TableGrid tables={eventRooms[currentEventRoom].meetingTables} />
                        <div className={classes.divider} />
                      </Grid>
                      <Grid item md={3} xs={12}>
                        <Typography variant="subtitle2">Scene: </Typography>
                        <Typography variant="body2">
                          {findScene(eventRooms[currentEventRoom].scene)}
                        </Typography>
                        <div className={classes.divider} />
                        <Typography variant="subtitle2">Background: </Typography>
                        <Typography variant="body2">
                          {isBackgroundPreset(eventRooms[currentEventRoom].background)
                            ? findBackground(eventRooms[currentEventRoom].background)
                            : getFileNameFromBlobUrl(eventRooms[currentEventRoom].background)}
                        </Typography>
                        <div className={classes.divider} />
                        <Typography variant="subtitle2">Event Room URL: </Typography>
                        <a
                          href={`${platformURL}?id=${eventRooms[currentEventRoom]._id}`}
                          target="_blank"
                          rel="noreferrer noopener"
                        >
                          <Typography variant="body2" noWrap>
                            {platformURL}?id={eventRooms[currentEventRoom]._id}
                          </Typography>
                        </a>
                        <Button
                          aria-label="copyUrl"
                          className={classes.mt}
                          variant="contained"
                          color="primary"
                          onClick={() => {
                            navigator.clipboard.writeText(
                              platformURL + "?id=" + eventRooms[currentEventRoom]._id
                            );
                          }}
                        >
                          Copy URL
                        </Button>
                      </Grid>
                    </Grid>
                  </CardContent>
                  <CardActions>
                    <Button
                      aria-label="editEventRoomButton"
                      color="primary"
                      variant="outlined"
                      onClick={() => {
                        handleClickEdit(currentEventRoom);
                      }}
                    >
                      Edit
                    </Button>
                    <Button
                      aria-label="removeEventRoomButton"
                      color="secondary"
                      variant="outlined"
                      onClick={() => {
                        handleClickRemove(eventRooms[currentEventRoom]._id);
                      }}
                    >
                      Remove
                    </Button>
                  </CardActions>
                </React.Fragment>
              ) : (
                <Typography id="createEventRoomButton" color="textSecondary" gutterBottom>
                  Create an event room
                </Typography>
              )}
            </Card>
          </Grid>
        </Grid>
      )}
    </Container>
  );
};

export default EventRooms;
