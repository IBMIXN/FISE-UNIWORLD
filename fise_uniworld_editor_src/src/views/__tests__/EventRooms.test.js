import React from "react";
import { shallow } from "enzyme";
import EventRooms from "../EventRooms";
import EventRoomService from "../../services/EventRoomService";
import { mockEventRoom, mockEventRoomUpdate } from "../../mockData/eventRoomMock";
import { useReducerState } from "../../utils/customHooks";
import Snackbar from "@material-ui/core/Snackbar";
import { platformURL } from "../../config";

jest.mock("../../services/EventRoomService");
jest.mock("../../utils/validate");
jest.mock("../../utils/customHooks", () => {
  return {
    useReducerState: jest.fn(),
  };
});

Object.assign(navigator, {
  clipboard: {
    writeText: () => {},
  },
});

describe("Eventrooms view test", () => {
  const location = { state: { created: false } };
  const match = { params: { id: "mockId" } };
  const history = { push: jest.fn() };

  const flushPromises = () => new Promise(setImmediate);
  const alert = {
    open: false,
    message: "",
    type: "success",
  };
  const setAlert = jest.fn();

  beforeEach(() => {
    useReducerState.mockImplementation(() => [alert, setAlert]);
    jest.spyOn(console, "log").mockImplementation(() => {});
    jest.spyOn(console, "warn").mockImplementation(() => {});
    window.confirm = jest.fn().mockImplementation(() => true);
    EventRoomService.getAll.mockImplementation(() =>
      Promise.resolve([{ ...mockEventRoom, _id: "mockId" }])
    );
  });

  it("should match snapshot", () => {
    const component = shallow(<EventRooms location={location} match={match} />);
    expect(component).toMatchSnapshot();
  });

  it("should match snapshot with event rooms", async () => {
    const component = shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    expect(component).toMatchSnapshot();
  });

  it("should fetch event rooms from EventRoomService on load", async () => {
    shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    expect(EventRoomService.getAll).toBeCalled();
  });

  it("should handle errors while fetching event rooms from EventRoomService on load", async () => {
    EventRoomService.getAll.mockImplementationOnce(() => Promise.reject("error"));
    shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    expect(EventRoomService.getAll).toBeCalled();
    expect(console.log).toBeCalledWith("error");
  });

  it("should show alert if location state created is true", async () => {
    shallow(<EventRooms location={{ state: { created: true } }} match={match} />);
    await flushPromises();
    expect(setAlert).toBeCalledWith({
      open: true,
      message: "Successfully created event room.",
      type: "success",
    });
  });

  it("should not show alert if location state is undefined", async () => {
    shallow(<EventRooms location={{}} match={match} />);
    await flushPromises();
    expect(setAlert).not.toBeCalled();
  });

  it("should correctly handle remove event room button click", async () => {
    const component = shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    expect(component.exists("#createEventRoomButton")).toEqual(false);
    component.find('[aria-label="removeEventRoomButton"]').first().simulate("click");
    await flushPromises();
    expect(component.exists("#createEventRoomButton")).toEqual(true);
    expect(window.confirm).toBeCalled();
    expect(EventRoomService.delete).toBeCalledWith("mockId");
  });

  it("should correctly handle errors during remove event room button click", async () => {
    EventRoomService.delete.mockImplementationOnce(() => Promise.reject("error"));
    const component = shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    component.find('[aria-label="removeEventRoomButton"]').first().simulate("click");
    await flushPromises();
    expect(component.exists("#createEventRoomButton")).toEqual(false);
    expect(window.confirm).toBeCalled();
    expect(EventRoomService.delete).toBeCalledWith("mockId");
    expect(console.log).toBeCalledWith("error");
  });

  it("should correctly handle edit event room button click", async () => {
    const component = shallow(<EventRooms history={history} location={location} match={match} />);
    await flushPromises();
    expect(component.exists("#createEventRoomButton")).toEqual(false);
    component.find('[aria-label="editEventRoomButton"]').first().simulate("click");
    await flushPromises();
    expect(history.push).toBeCalledWith({
      pathname: "/edit",
      state: { eventRoom: { ...mockEventRoom, _id: "mockId" } },
    });
  });

  it("should correctly handle close alert dialog", () => {
    const component = shallow(<EventRooms location={location} match={match} />);
    component.find(Snackbar).first().prop("onClose")();
    expect(setAlert).toBeCalledWith({ open: false });
  });

  it("should correctly select event room from list", async () => {
    EventRoomService.getAll.mockImplementationOnce(() =>
      Promise.resolve([
        { ...mockEventRoom, _id: "mockId" },
        { ...mockEventRoomUpdate, _id: "mockId2" },
      ])
    );
    const component = shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    expect(component.find("#currentEventRoomTitle").first().text()).toEqual(mockEventRoom.title);
    expect(component.exists("#listItem0")).toEqual(true);
    expect(component.exists("#listItem1")).toEqual(true);
    expect(component.find("#listItem1").first().prop("selected")).toEqual(false);
    component.find("#listItem1").first().simulate("click");
    expect(component.find("#listItem1").first().prop("selected")).toEqual(true);
    expect(component.find("#currentEventRoomTitle").first().text()).toEqual(
      mockEventRoomUpdate.title
    );
  });

  it("should correctly handle copying url", async () => {
    jest.spyOn(navigator.clipboard, "writeText");
    const component = shallow(<EventRooms location={location} match={match} />);
    await flushPromises();
    expect(component.exists('[aria-label="copyUrl"]')).toEqual(true);
    component.find('[aria-label="copyUrl"]').first().simulate("click");
    expect(navigator.clipboard.writeText).toBeCalledWith(`${platformURL}?id=mockId`);
  });
});
