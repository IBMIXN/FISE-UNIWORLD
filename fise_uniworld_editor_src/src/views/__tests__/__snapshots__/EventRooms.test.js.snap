// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eventrooms view test should match snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-mainSection-1"
>
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
    autoHideDuration={3000}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(Alert))
      elevation={6}
      onClose={[Function]}
      severity="success"
      variant="filled"
    />
  </WithStyles(ForwardRef(Snackbar))>
  <WithStyles(ForwardRef(Typography))
    variant="h4"
  >
    Event Rooms
  </WithStyles(ForwardRef(Typography))>
  <div
    className="makeStyles-progressLoader-5"
  >
    <WithStyles(ForwardRef(CircularProgress)) />
  </div>
</WithStyles(ForwardRef(Container))>
`;

exports[`Eventrooms view test should match snapshot with event rooms 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-mainSection-1"
>
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
    autoHideDuration={3000}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(Alert))
      elevation={6}
      onClose={[Function]}
      severity="success"
      variant="filled"
    />
  </WithStyles(ForwardRef(Snackbar))>
  <WithStyles(ForwardRef(Typography))
    variant="h4"
  >
    Event Rooms
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={3}
      xs={12}
    >
      <WithStyles(ForwardRef(List))
        aria-label="event rooms"
        component="nav"
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
          className="makeStyles-listItemCreate-6"
          component={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          to="/create"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <Memo([object Object])
              style={
                Object {
                  "color": "#fff",
                }
              }
            />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Create event room"
            primaryTypographyProps={
              Object {
                "noWrap": true,
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          id="listItem0"
          key="mockId"
          onClick={[Function]}
          selected={true}
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <Memo([object Object]) />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="mockTitle"
            primaryTypographyProps={
              Object {
                "noWrap": true,
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
      </WithStyles(ForwardRef(List))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      md={9}
      xs={12}
    >
      <WithStyles(ForwardRef(Card))
        className="makeStyles-eventInfoPanel-2"
      >
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
          >
            March 17 2021, 12:00 am
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            component="h2"
            id="currentEventRoomTitle"
            variant="h5"
          >
            mockTitle
          </WithStyles(ForwardRef(Typography))>
          <div
            className="makeStyles-divider-3"
          />
          <WithStyles(ForwardRef(Grid))
            container={true}
          >
            <WithStyles(ForwardRef(Grid))
              className="makeStyles-centerTableGrid-4"
              item={true}
              md={9}
              xs={12}
            >
              <TableGrid
                tables={Array []}
              />
              <div
                className="makeStyles-divider-3"
              />
            </WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Grid))
              item={true}
              md={3}
              xs={12}
            >
              <WithStyles(ForwardRef(Typography))
                variant="subtitle2"
              >
                Scene: 
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                variant="body2"
              >
                mockScene
              </WithStyles(ForwardRef(Typography))>
              <div
                className="makeStyles-divider-3"
              />
              <WithStyles(ForwardRef(Typography))
                variant="subtitle2"
              >
                Background: 
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Typography))
                variant="body2"
              />
              <div
                className="makeStyles-divider-3"
              />
              <WithStyles(ForwardRef(Typography))
                variant="subtitle2"
              >
                Event Room URL: 
              </WithStyles(ForwardRef(Typography))>
              <a
                href="https://brianmin.tech/uniworld?id=mockId"
                rel="noreferrer noopener"
                target="_blank"
              >
                <WithStyles(ForwardRef(Typography))
                  noWrap={true}
                  variant="body2"
                >
                  https://brianmin.tech/uniworld
                  ?id=
                  mockId
                </WithStyles(ForwardRef(Typography))>
              </a>
              <WithStyles(ForwardRef(Button))
                aria-label="copyUrl"
                className="makeStyles-mt-7"
                color="primary"
                onClick={[Function]}
                variant="contained"
              >
                Copy URL
              </WithStyles(ForwardRef(Button))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(CardActions))>
          <WithStyles(ForwardRef(Button))
            aria-label="editEventRoomButton"
            color="primary"
            onClick={[Function]}
            variant="outlined"
          >
            Edit
          </WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Button))
            aria-label="removeEventRoomButton"
            color="secondary"
            onClick={[Function]}
            variant="outlined"
          >
            Remove
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(CardActions))>
      </WithStyles(ForwardRef(Card))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
